doc/languages-frameworks/lua.md

```diff
@@ -0,0 +1,262 @@
+to go forward and start actually developing with Lua. There are two ways to
+package lua software, either it is on luarocks and most of it can be taken care
+of by the luarocks2nix converter or the packaging has to be done manually.
+Let's present the luarocks way first and the manual one in a second time.
+
+### Packaging a library on luarocks
+
+Luarocks.org is the main repository of lua packages. The site proposes two types of packages, the rockspec and the src.rock (equivalent of a rockspec but with the source). These packages can have different build types such as `cmake`, `builtin` etc (See https://github.com/luarocks/luarocks/wiki/Rockspec-format).
+
+A list of luarocks generated packages is visible at pkgs/top-level/lua-generated-packages.nix.
+
+
+Luarocks2nix is a tool capable of generating nix derivations from a rockspec file but there are at the moment some limits
+1. The nix derivation will try to fetch the sources from the src.rock file.
+   This can be worked around by using `luarocks pack <PKG>.rockspec` and then
+   upload the file on luarocks.org.
+2. nix won't work with all packages. If the package lists `external_dependencies` in its rockspec file then it won't work.
+
+You can run `nix-shell -p luarocks-nix` and then `luarocks convert2nix
+PKG_NAME`. Once you have checked the package works without modifications, you can add it to `maintainers/scripts/update-luarocks-packages.sh`
+Nix rely on luarocks to install lua packages, basically it runs:
+`luarocks make --deps-mode=none --tree $out`
+
+#### Packaging a library manually
+
+You can develop your package as you usually would, just don't forget to wrap it
+within a `toLuaModule` call.
+
+
+
+## Lua Reference
+
+### Lua interpreters
+
+Versions 5.1, 5.2 and 5.3 of the lua interpreter are available as
+respectively `lua_51`, `lua_52` and `lua_53`.
+The default interpreter, `lua`, maps to `lua_51`. The luajit interpreter is also available.
+The Nix expressions for the interpreters can be found in
+`pkgs/development/interpreters/lua`.
+
+All packages depending on any Lua interpreter get appended
+`out/share/{lua.version}/?.lua` to `$LUA_PATH` if such directory
+exists.
+
+#### Attributes on lua interpreters packages
+
+Each interpreter has the following attributes:
+
+- `interpreter`. Alias for `${lua}/bin/lua`.
+- `buildEnv`. Function to build python interpreter environments with extra packages bundled together. See section *python.buildEnv function* for usage and documentation.
+- `withPackages`. Simpler interface to `buildEnv`.
+- `pkgs`. Set of Lua packages for that specific interpreter. The package set can be modified by overriding the interpreter and passing `packageOverrides`.
+
+
+#### `buildLuaPackage` function
+
+The `buildLuaPackage` function is implemented in `pkgs/development/interpreters/lua-5/build-lua-package.nix`
+The following is an example:
+```nix
+
+buildLuaPackage rec {
+  version = "3.3.1";
+  pname = "pytest";
```

Comment:
That one should be replaced by a proper lua example.

Reply:
thanks for looking at it. I pushed a fix. Note that I will improve the commit message once all reviews are adressed (rebase + autosquash sometimes trigger conflicts so I end up prefering to soft reset and copy/rewrite commit messages).